File: lib/domain/crop/entities/thermal_requirement_entity.rb
# frozen_string_literal: true

module Domain
  module Crop
    module Entities
      class ThermalRequirementEntity
        attr_reader :id, :crop_stage_id, :required_gdd

        def initialize(attributes)
          @id = attributes[:id]
          @crop_stage_id = attributes[:crop_stage_id]
          @required_gdd = attributes[:required_gdd]

          validate!
        end

        # ActiveRecordモデルからの変換
        def self.from_model(model)
          new(
            id: model.id,
            crop_stage_id: model.crop_stage_id,
            required_gdd: model.required_gdd
          )
        end

        private

        def validate!
          raise ArgumentError, "Crop stage ID is required" if crop_stage_id.blank?
        end
      end
    end
  end
end

File: crates/agrr-domain/src/crop/entities/thermal_requirement_entity.rs
use rust_decimal::Decimal;

/// Ruby: `Domain::Crop::Entities::ThermalRequirementEntity`
#[derive(Debug, Clone, PartialEq)]
pub struct ThermalRequirementEntity {
    pub id: i64,
    pub crop_stage_id: i64,
    pub required_gdd: Decimal,
}

impl ThermalRequirementEntity {
    pub fn new(id: i64, crop_stage_id: i64, required_gdd: Decimal) -> Result<Self, String> {
        if crop_stage_id == 0 {
            return Err("Crop stage ID is required".into());
        }
        Ok(Self {
            id,
            crop_stage_id,
            required_gdd,
        })
    }
}
